client/sites/components/site-preview-pane/dotcom-preview-pane.tsx

```diff
@@ -188,6 +189,18 @@ const DotcomPreviewPane = ( {
 		stagingStatus === StagingSiteStatus.NONE ||
 		stagingStatus === StagingSiteStatus.UNSET;
 
+	const hasStagingSitePermission = stagingSites?.some(
+		( stagingSite ) => stagingSite.user_has_permission
+	);
+
+	const hasProductionSitePermission = useSelector( ( state ) => {
```

Comment:
This is probably not the best way to check if the user has permission for production.
Please let me know if you have a better suggestion.

Reply:
Code LGTM but I feel some naming confusion:
1. we have `hasStagingSitePermission`
2. we have `hasProductionSitePermission`, but inside of it we have `site.is_wpcom_staging_site`, but staging site was checked with `hasStagingSitePermission`.

Feels like we distinguish staging/production with two different variables but at the same time we also check staging inside production.